### DISCUSSION TITLE
GitHub consuming 100% of CPU core, won't relinquish unless entire browser closed

### DISCUSSION THREAD

**Discussion by Ben-Fenner** [community/community#23906] [New to GitHub]
<p>I’m running the latest Linux Mint (20.1 64-bit) and using a Firefox-forked browser.<br>
Scrolling nearly any page on GitHub immediately gives the console error seen in the screenshot below, and then 100% of one CPU core is taken up by the browser process. The CPU core is not relinquished until the browser is closed (closing the tab is not enough).</p>
<p>Someone who knows better than me on the topic suggests “[Web devs] should stop manually repositioning elements with JS upon scrolling because it’s very CPU intensive, and use the CSS keywords for it that have been around for a looooong time.”</p>
<p>Here is the URL in the error message so no one has to re-type it: <a href="https://developer.mozilla.org/en-US/docs/Mozilla/Performance/Scroll-linked_effects" class="inline-onebox" rel="noopener nofollow ugc">Scroll-linked effects - Mozilla | MDN</a></p>
<p>Let me know if I can provide more details.</p>
<p>Screenshot:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/c/7ca63a9ec1cb208aa92562be75a75d3e79ecb5ee.png" data-download-href="/uploads/short-url/hMHlIwIWg8WlzXotKJg0HJ1V5H8.png?dl=1" title="" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/75506118/181079929-9412e0cc-2101-4670-9949-f34ce91243b0.png" alt="" data-base62-sha1="hMHlIwIWg8WlzXotKJg0HJ1V5H8" width="652" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/c/7ca63a9ec1cb208aa92562be75a75d3e79ecb5ee_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename"></span><span class="informations">716×549 31.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Ben-Fenner** [comment]
<p>I’ve done some more testing and it doesn’t seem related to scrolling anymore, so I’ve updated the title.</p>
<p>I was able to reproduce the issue by opening a single browser tab/window to our GitHub issues page that was never scrolled and has no scrollbar. After leaving the window/tab open for ~5 minutes (and moving the mouse over the window a bit toward the end) the CPU core spiked and we were off to the races. The developer console showed zero output this time.</p>
<p>I’m not sure what is going on, but I assume some JS infinite loop?</p>
<p>Let me know if I can help diagnose further.</p>

**Comment by canuckjacq** [comment]
<p>Are you using Dark Mode? I see there’s a similar report here: <a href="https://github.community/t/about-perfomance/170052" class="inline-onebox">About perfomance</a></p>

**Comment by Ben-Fenner** [comment]
<p>Thanks for the suggestion, but no I have not enabled dark mode. =/</p>

**Comment by Ben-Fenner** [comment]
<p>Is there any more input on the matter I can give? Anything else I can do to help diagnose?<br>
This is a daily/hourly annoyance I would love to avoid if possible.</p>

**Comment by canuckjacq** [comment]
<p>I don’t really have anything to be honest. I am seeing a lot of reports of Waterfox not working with GitHub but unfortunately that is a Waterfox issue and not one of our own. Is that what you’re using?</p>

**Comment by jfullstackdev** [comment]
<p>I am using a Celeron processor, from the series of HP Stream 5 years ago. The Operating System is Windows 8.1. I’m using Google Chrome as the browser. I’m also checking the Task Manager from time to time just to see the stats. Together with GitHub (even if I’m using Dark Mode), GitHub Community, Facebook and Twitter, the stats is really good.</p>
<p>Maybe, it’s really your machine and the installed apps, not GitHub.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/e/1ed89840354ef87ec388dec169b4c4d36144f002.png" data-download-href="/uploads/short-url/4oSnk2DZQavGlfw0puKkteE2mfo.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181079934-da656747-176e-4a2f-9054-e873acaa10ae.png" alt="image" data-base62-sha1="4oSnk2DZQavGlfw0puKkteE2mfo" width="690" height="387" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/e/1ed89840354ef87ec388dec169b4c4d36144f002_2_690x387.png, https://aws1.discourse-cdn.com/github/optimized/3X/1/e/1ed89840354ef87ec388dec169b4c4d36144f002_2_1035x580.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/e/1ed89840354ef87ec388dec169b4c4d36144f002.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/e/1ed89840354ef87ec388dec169b4c4d36144f002_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1366×768 93.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Ben-Fenner** [comment]
<p>canuckjacq, I am using Pale Moon, not Waterfox. What is the issue Waterfox has?</p>
<p>xdvrx1, you’re not using a standards compliant browser, so I’m not surprised you’re not having problems. It is most assuredly a combination of GitHub, and my standards compliant browser that is causing an issue. And if history is anything to go by, it is almost always the web site causing the issue, not the browser.<br>
But the browser developers are available to help diagnose and fix the issue on either end if the GitHub developers would like to reach out to them on their forum. Or probably vise versa.</p>
<p>If you’d like to join the discussion on the Pale Moon forum, here is that link: <a href="https://forum.palemoon.org/viewtopic.php?f=37&amp;t=26322#p209996" class="inline-onebox" rel="noopener nofollow ugc">Pale Moon (Linux) consuming 100% of CPU core, won't relinquish unless app closed. - Pale Moon forum</a></p>
<p>By the way, this does not happen on the Pale Moon browser on any of my four Windows 7 machines, only on this one Linux machine. So I’m sure that plays into it in some way.</p>

**Comment by jfullstackdev** [comment]
<aside class="quote no-group" data-username="Ben-Fenner" data-post="8" data-topic="168703">
<div class="title">
<div class="quote-controls"></div>
 Ben-Fenner:</div>
<blockquote>
<p>By the way, this does not happen on the Pale Moon browser on any of my four Windows 7 machines, only on this one Linux machine. So I’m sure that plays into it in some way.</p>
</blockquote>
</aside>
<p>yup, that’s what I’m saying too, the machine itself is one factor so I posted my specifics</p>

**Comment by canuckjacq** [comment]
<blockquote>
<p>canuckjacq, I am using Pale Moon, not Waterfox. What is the issue Waterfox has?</p>
</blockquote>
<p>I think (this is not my area of expertise) the lack of Web Components support is the big blocker. I suspect the polyfill add on is the same for both Waterfox Classic and Pale Moon since they’re both forked from older Firefox ESR.</p>
<p>The Pale Moon community probably knows better and there’s a thread here: <a href="https://forum.palemoon.org/viewtopic.php?f=70&amp;t=25435&amp;start=20" class="inline-onebox">PM 28.14.2 Github - Page 2 - Pale Moon forum</a></p>
<p>We do have a list of our <a href="https://docs.github.com/en/github/getting-started-with-github/supported-browsers">supported browsers</a> available for full feature compatibility.</p>

**Comment by Ben-Fenner** [comment]
<p>Yes, you’re quite likely right the lack of WebComponents support in Pale Moon is the issue (or GitHub’s insistence on using WebComponents…). The polyfill add-on I’m using to even get GitHub to work at all in Pale Moon has been a life saver, as I would not be using GitHub without it.</p>
<p>That list of supported browsers reads like a nightmare, and building to those non-standards compliant browsers is not something this web developer would ever recommend. But I digress.</p>
<p>It seems I’ll either have to wait for GitHub to build and test against standards compliant browsers, or wait for Pale Moon to gain WebComponents capability natively, or live with this poor behavior, or start moving our shop away from GitHub.</p>
<p>Cheers!</p>

**Comment by Ben-Fenner** [comment]
<p>It is possible this issue no longer exists. I’m not sure if GitHub is behaving better or what, but I’ve had a couple GitHub tabs open for a few days now and still have normal CPU usage. This would have been impossible before.</p>

**Comment by Ben-Fenner** [comment, marked as answer]
<p>It has been more than a month and this issue has not cropped up again.<br>
I’m marking this “solved”.</p>

**Comment by mmospanenko** [comment]
it is a spinner. Hide (switch on another tab, window) it and loading reduces. CSS animation dries battery... heats the Earth :)
<img width="77" alt="image" src="https://github.com/community/community/assets/2766211/388ab0a4-dd51-48d4-9f13-15245086495b">

**Reply by shalom938** [reply]
I have the same issue on linux with vivaldi browser. its definitely the spinner  using high cpu

**Reply by mmospanenko** [reply]
I fixed with disabling animation:

<img width="280" alt="image" src="https://github.com/community/community/assets/2766211/bf15c109-1d37-4f4d-a5f7-1d771cfc73f0">
